cudapoa/src/cudapoa_kernels.cuh

```diff
@@ -221,12 +235,12 @@ __launch_bounds__(GW_POA_KERNELS_MAX_THREADS_PER_BLOCK)
 
     __syncwarp();
 
-    // Align each subsequent read, add alignment to graph, run topoligical sort.
+    // Align each subsequent read, add alignment to graph, run topological sort.
     for (int32_t s = 1; s < num_sequences; s++)
     {
         int32_t seq_len = sequence_lengths[s];
-        sequence += sequence_lengths[s - 1];     // increment the pointer so it is pointing to correct sequence data
-        base_weights += sequence_lengths[s - 1]; // increment the pointer so it is pointing to correct sequence data
+        sequence += cudautils::align<int32_t, SIZE_OF_SeqT4>(sequence_lengths[s - 1]);     // increment the pointer so it is pointing to correct sequence data
```

Comment:
I think it should be both on host and device side. There are 2 parameters used in copying reads to host, then device: `sequence_lengths` and `num_nucleotides_copied`, the former is the actual number of bases which may or may not be aligned with 4. The latter determines space allocated for the copy which is aligned with 4: [see L538 and L541](https://github.com/clara-parabricks/GenomeWorks/blob/c1f69e9d1a9b2e28d96d3b44e822721e69337bda/cudapoa/src/cudapoa_batch.cuh#L538).
btw, changed `SIZE_OF_SeqT4` from 32 to 4.

Reply:
can we not use the `num_nucleotides_copied` buffer entry to find the start position of the buffer?

Reply:
`num_nucleotides_copied` is not being stored in any buffer and also cannot replace `sequence_lengths` buffer, since actual sequence length is needed in `NW`. So the solution is either allocate another buffer to store `num_nucleotides_copied` or compute it based on `sequence_lengths`. In the interest of memory, the second option is used here.

Reply:
Ah I was under the impression we were storing the offsets in a buffer already. My main concern is that the aligned offset calculation is done independently for the host and device side, so if any one of them is changed and the other isn't, it would lead to hard to find bugs. would it be easier if we got rid of the packed storage of reads and just indexed them with `sequences[window_idx * MAX_SEQUENCE_SIZE]`?

Reply:
that'll be wasteful as well, specially for unfiltered cases where variation of sequences within a group can be large. To somewhat make it traceable, I defined the macro `SIZE_OF_SeqT4` where is being used in 2 places in the code, one is in the host, the second here. But if you think that can be overlooked in the future, an alternative is simply remove `__align__` from `SeqT4` in `cudapoa_structs.cuh`, as it was in the past, and just work with actual length of sequences without imposing alignment. This will be sub-optimal but doubt it would have a big impact on performance.

Reply:
yes, fair enough. let's keep the `__align__` in the struct since it optimizes read performance from global memory. and TODO is to add a comment mentioning that the alignment is done separately on host and device side, as a heads up to future developers.